tests/data/config/invalid_bad_max_daz.yaml

```diff
@@ -0,0 +1,3 @@
+algorithm_name: "simple"
```

Comment:
I'm not sure about having invalid configuration with the code. I think that this should cause a unit test to fail so we would have an easy way to catch malformed configurations.

Reply:
These configurations are in tests/data/config which is intended solely for unit tests.

I don't anticipate anyone running the CSC with this as the real configuration directory because configuration files are intended to live in a different package. (I would be much more worried about this if the "config" directory was at the top level instead of down in "tests/data").

Or perhaps I am missing your point?

Reply:
Oh, the path of the file completely escaped my attention. I think it is ok to have it here for test-cases. 

I think it is ok that `TestCsc` overwrites `config_dir` but maybe it needs some explanation there that this should be avoided in real CSCs? 

But then, `ATDomeTrajectory` is pointing to the configuration in `<ts_salobj_root>/tests/data/config` by default.  Maybe that needs to be changed and introduce an external configuration repo?

Reply:
`ATDomeTrajectory` has no default for `config_dir`, or am I missing something?

`TestCsc` does have a default and I agree I should document that real CSCs should not do that or else make it like a real CSC and fix ts_DockerFiles on this ticket branch -- which has a certain appeal.

Reply:
You are right, but the documentation said otherwise:

https://github.com/lsst-ts/ts_ATDomeTrajectory/blob/9620a3a547488322ec1fc979531d69397a260df9/python/lsst/ts/ATDomeTrajectory/dome_trajectory.py#L46-L48

Reply:
Oops. I'll fix that. This is a good argument in favor of making `TestCsc` *not* have a default `config_dir`!

Reply:
The config_dir is now optional for all configurable CSCs.